rest.go

```diff
@@ -90,39 +51,39 @@ func BuildResponse(res *http.Response) (*Response, error) {
 }
 
 // API is the main interface to the API.
-func API(request Request) (*Response, error) {
+func API(request *http.Request) (*Response, error) {
```

Comment:
I agree. In the initial request it only really spoke to using http.Request proper but it does seem to make more sense to keep everything congruent.